### PR TITLE
Leader election only uses Lease object

### PR DESCRIPTION
**Pull request by sunjayBhatia**
- Removes RBAC for ConfigMaps
- Also removes mentions of leader election ConfigMaps from examples
- Updates e2e tests

Fixes: #1393

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
the config file still has leaderelection configuration here: https://projectcontour.io/docs/main/configuration/#leader-election-configuration

and it mentions a configmap

we note that it is "deprecated" but when can we remove that?

**Comment by sunjayBhatia** [comment]
will need to also update the operator once this merges

**Comment by skriss** [comment]
> the config file still has leaderelection configuration here: https://projectcontour.io/docs/main/configuration/#leader-election-configuration
> 
> and it mentions a configmap
> 
> we noted last release that it is "deprecated" but when can we remove the option entirely?

https://projectcontour.io/resources/deprecation-policy/ does have some general guidance ("choose your own timeline"), in this case IMHO it's fine to drop in 1.21 as long as we explicitly call it out in the release notes. Maybe a good time to get #4258 going so we don't forget to call it out?

**Comment by sunjayBhatia** [comment]
> > the config file still has leaderelection configuration here: https://projectcontour.io/docs/main/configuration/#leader-election-configuration
> > and it mentions a configmap
> > we noted last release that it is "deprecated" but when can we remove the option entirely?
> 
> https://projectcontour.io/resources/deprecation-policy/ does have some general guidance ("choose your own timeline"), in this case IMHO it's fine to drop in 1.21 as long as we explicitly call it out in the release notes. Maybe a good time to get #4258 going so we don't forget to call it out?

yeah https://github.com/projectcontour/contour/releases/tag/v1.20.0 calls it out as deprecated, though I didn't add a timeline in that release note

I'll work on the deprecations/removals release note tag, thats good to have

**Comment by skriss** [comment]
I'll fold the RBAC update into https://github.com/projectcontour/contour-operator/pull/487 as well so we don't need another PR

**Review by skriss — APPROVED** [review]
LGTM but fine if you also want to drop the config file fields in this PR

**Comment by sunjayBhatia** [comment]
> LGTM but fine if you also want to drop the config file fields in this PR

I'll do that in a separate PR once https://github.com/projectcontour/contour/pull/4333 is merged to ensure it all works

**Pull request merged**
